In [1]:
import pandas as pd
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import train_test_split
from sklearn.pipeline import Pipeline
import numpy as np
import matplotlib.pyplot as plt

In [5]:
# Load Datasets

train = pd.read_csv('../data/train.csv')
test = pd.read_csv('../data/test.csv')

In [8]:
train.head(2).T

0                 1
rent_approval_date                 2021-09           2022-05
town                           jurong east             bedok
block                                  257               119
street_name          Jurong East Street 24  bedok north road
flat_type                           3 room            4-room
flat_model                  new generation    new generation
floor_area_sqm                        67.0              92.0
furnished                              yes               yes
lease_commence_date                   1983              1978
latitude                          1.344518          1.330186
longitude                        103.73863        103.938717
elevation                              0.0               0.0
subzone                         yuhua east       bedok north
planning_area                  jurong east             bedok
region                         west region       east region
monthly_rent                          1600              2250

In [11]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60000 entries, 0 to 59999
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   rent_approval_date   60000 non-null  object 
 1   town                 60000 non-null  object 
 2   block                60000 non-null  object 
 3   street_name          60000 non-null  object 
 4   flat_type            60000 non-null  object 
 5   flat_model           60000 non-null  object 
 6   floor_area_sqm       60000 non-null  float64
 7   furnished            60000 non-null  object 
 8   lease_commence_date  60000 non-null  int64  
 9   latitude             60000 non-null  float64
 10  longitude            60000 non-null  float64
 11  elevation            60000 non-null  float64
 12  subzone              60000 non-null  object 
 13  planning_area        60000 non-null  object 
 14  region               60000 non-null  object 
 15  monthly_rent         60000 non-null 

In [13]:
train['rent_approval_date'] = pd.to_datetime(train['rent_approval_date'])

In [15]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60000 entries, 0 to 59999
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   rent_approval_date   60000 non-null  datetime64[ns]
 1   town                 60000 non-null  object        
 2   block                60000 non-null  object        
 3   street_name          60000 non-null  object        
 4   flat_type            60000 non-null  object        
 5   flat_model           60000 non-null  object        
 6   floor_area_sqm       60000 non-null  float64       
 7   furnished            60000 non-null  object        
 8   lease_commence_date  60000 non-null  int64         
 9   latitude             60000 non-null  float64       
 10  longitude            60000 non-null  float64       
 11  elevation            60000 non-null  float64       
 12  subzone              60000 non-null  object        
 13  planning_area        60000 non-

street_name	Jurong East Street 24	bedok north road	lorong 1 toa payoh	Pasir Ris Street 21	Whampoa West
flat_type	3	4	3	6	3
flat_model	2370	2370	2636	2879	2636
floor_area_sqm	67.0	92.0	67.0	149.0	68.0
lease_commence_date	1983	1978	1971	1993	1972
latitude	1.344518	1.330186	1.332242	1.370239	1.320502
longitude	103.73863	103.938717	103.845643	103.962894	103.863341
subzone	yuhua east	bedok north	toa payoh central	pasir ris drive	bendemeer
planning_area	jurong east	bedok	toa payoh	pasir ris	kallang
region	west region	east region	central region	east region	central region
monthly_rent	1600	2250	1900	2850	2100
dist_mrt_exist	0.006289	0.008087	0.001966	0.013908	0.00169
dist_mrt_planned	0.006071	0.008135	0.010649	0.004643	0.025345
dist_primary_school	